File: backend/model/problem.go
package model

import (
	"gorm.io/gorm"
)

type Problem struct {
	gorm.Model
	Title             string             `json:"title" gorm:"type:varchar(64); unique"`
	UserId            int64              `json:"user_id" gorm:"type:int;"`
	Description       string             `json:"description" gorm:"type:text;not null"`
	Input             string             `json:"input" gorm:"type:text;not null"`
	Output            string             `json:"output" gorm:"type:text;not null"`
	SimpleInput       string             `json:"simple_input" gorm:"type:text"`
	SimpleOutput      string             `json:"simple_output" gorm:"type:text"`
	Illustrate        string             `json:"illustrate" gorm:"type:text"`
	Data              string             `json:"data" gorm:"type:varchar(64);not null"`
	TimeLimit         uint32             `json:"time_limit" gorm:"type:int"`
	MemoLimit         uint32             `json:"memo_limit" gorm:"type:int"`
	ProblemCategories []*ProblemCategory `gorm:"foreignKey:problem_id;references:id"`
}

File: frontend/src/store/module/problem.js
import problemService from '@/service/problemService';


const problemModule = {
    namespaced: true,

    state: {
        problemList: [],
        problemDetail: '',
    },
    mutations: {
        setProblemList(state, problemList) {
            state.problemList = problemList;
        },
        setProblemDetail(state, problemDetail) {
            state.problemDetail = problemDetail;
        },
    },
    actions: {
        async add({ commit }, { Title, UserId, Description, Input, Output, SimpleInput, SimpleOutput, Illustrate, TimeLimit, MemoLimit }) {
            try {
                const res = await problemService.add({ Title, UserId, Description, Input, Output, SimpleInput, SimpleOutput, Illustrate, TimeLimit, MemoLimit });
                return res;
            } catch (error) {
                // 在这里处理错误，比如打印错误日志或通知用户  
                console.error('Error adding problem:', error);
                throw error; // 重新抛出错误，以便在调用处能够捕获到  
            }
        },
        async fetchProblemList({ commit }) {
            try {
                const queryParams = window.location.search;
                const response = await problemService.GetProblemList(queryParams);
                const problemList = response.data;
                commit('setProblemList', problemList);
            } catch (error) {
                console.error('Request failed:', error);
                throw error;
            }
        },
        async GetProblemDetail({ commit }) {
            try {
                const queryParams = window.location.search;
                // 将查询参数转换为 URL 搜索参数字符串  
                const response = await problemService.GetProblem(queryParams);
                const problemDetail = response.data;
                commit('setProblemDetail', problemDetail);
            } catch (error) {
                // 处理请求错误  
                console.error('Request failed:', error);
                throw error;
            }
        },
        async EditProblem({ commit }, { Title, UserId, Description, Input, Output, SimpleInput, SimpleOutput, Illustrate, TimeLimit, MemoLimit }) {
            try {
                const queryParams = window.location.search;
                const response = await problemService.edit(queryParams, { Title, UserId, Description, Input, Output, SimpleInput, SimpleOutput, Illustrate, TimeLimit, MemoLimit });
                return response;
            } catch (error) {
                // 在这里处理错误，比如打印错误日志或通知用户  
                console.error('Error adding problem:', error);
                throw error; // 重新抛出错误，以便在调用处能够捕获到  
            }
        },
    },
};

export default problemModule;
